### PR TITLE
feat: implement league promotion and demotion processing

### LINKED ISSUE #701: Implement league promotion and demotion logic

### PR DESCRIPTION
**Pull request by Inbi-Nav**
**Description**

Implemented weekly league promotion and demotion processing.
**Included**

* Added ProcessLeaguePromotions artisan command
* Implemented promotion and demotion logic between leagues
* Added protection for minimum and maximum league boundaries
* Prevented double promotions/demotions during the same execution
* Added feature tests for promotions, demotions, and edge cases

### CODE REVIEW

**Review by H3llynx — APPROVED** [review]
Maybe a bit long but indeed necessary. Great job!

**Review by orlandorojasoyarzun — APPROVED** [review]
Great job on implementing the logic !

I will also agree with @H3llynx that is still a bit too long but at this point, is hard to modularize it more, considering the test must be added to the feat. Let's hope for the best because the implementation is well structure and solid itself !

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
I would split this into:

1. Basic promotion case (best 3 move up!)
2. Basic demotion case (worst 3 move down!)
3. Edge case: cannot promote in gold / first league
4. Edge case: cannot demote in bronze / last league
5. Edge case: no promotion or demotion in leagues under 3 users

**Comment by Inbi-Nav** [comment]
> I would split this into:
> 
> 1. Basic promotion case (best 3 move up!)
> 2. Basic demotion case (worst 3 move down!)
> 3. Edge case: cannot promote in gold / first league
> 4. Edge case: cannot demote in bronze / last league
> 5. Edge case: no promotion or demotion in leagues under 3 users

Thanks for pointing it out!  I've updated it for better separation

File: backend/src/tests/Feature/LigaTests/LigaPromotionTest.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public function test_top_3_users_get_promoted(): void {
+        $users = User::factory()->count(9)->create();
+        for ($i = 0; $i < 9; $i++) {
+        Liga::create(['user_id' => $users[$i]->id, 'points_weekly' => 100 - ($i * 10), 'league_id' => 1 ]);
+        }
+        $this->artisan('liga:process-promotions')->assertSuccessful();
+        $this->assertEquals(2, Liga::where('user_id', $users[0]->id)->first()->league_id);
+        $this->assertEquals(2, Liga::where('user_id', $users[1]->id)->first()->league_id);
+        $this->assertEquals(2, Liga::where('user_id', $users[2]->id)->first()->league_id);
+    }
```
Comment (kenan-it-academy) [inline backend/src/tests/Feature/LigaTests/LigaPromotionTest.php:28]: please put newlines to separate tests for readability

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
@Inbi-Nav I meant to split this in different PRs, of which I see the five listed ones, not a separation within this PR

File: backend/src/tests/Feature/LigaTests/LigaPromotionTest.php (hunk 1 above)
Comment (Inbi-Nav) [reply backend/src/tests/Feature/LigaTests/LigaPromotionTest.php:28]: Done

**Comment by Inbi-Nav** [comment]
@kenan-it-academy need your review....
